File: FestivalC#/Festival/Networking/rpcprotocol/Request.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Networking.rpcprotocol
{
    [Serializable]
    public class Request
    {
        //type of request
        private RequestType type;

        //data sent
        private object data;

        public RequestType Type
        {
            get
            {
                return type;
            }

            set
            {
                type = value;
            }
        }

        public object Data
        {
            get
            {
                return data;
            }

            set
            {
                data = value;
            }
        }

        public Request(RequestType type, object data)
        {
            this.type = type;
            this.data = data;
        }

        public override string ToString()
        {
            return "Request{ type = " + Type + " data = " + Data + " } ";
        }

    }
}

File: FestivalJava/Networking/src/main/java/rpcprotocol/Request.java
package rpcprotocol;

import java.io.Serializable;

/**
 * Created by Sebi on 29-Mar-17.
 */
public class Request implements Serializable {
    //type of request
    private RequestType type;

    //data sent
    private Object data;

    //private constructor
    private Request(){}

    //getters and setters
    public RequestType getType(){
        return type;
    }

    private void setType(RequestType type){
        this.type = type;
    }

    public Object getData(){
        return data;
    }

    public void setData(Object data){
        this.data = data;
    }

    @Override
    public String toString(){
        return "Request{ type = " + getType() + " data = " + getData() + " } ";
    }

    public static class Builder{
        private Request request = new Request();

        public Builder type(RequestType type){
            request.setType(type);
            return this;
        }

        public Builder data(Object data){
            request.setData(data);
            return this;
        }

        public Request build(){
            return request;
        }
    }


}
